### PR TITLE
BlogPost: Collecting file based Java logs with OpenTelemetry

### PR DESCRIPTION
**Pull request by cyrille-leclerc**

### CODE REVIEW

**Comment by cyrille-leclerc** [comment]
cc @zeitlinger

File: content/en/blog/2024/collecting-file-based-java-logs-with-opentelemetry.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+    # TODO add "traces" and "metrics" pipelines
+
+exporters:
+  debug:
+    verbosity: detailed
+  # Exporter to the OTLP backend like `otlphttp`
+  otlphttp:
+
+```
+
```
Comment (cyrille-leclerc) [inline content/en/blog/2024/collecting-file-based-java-logs-with-opentelemetry.md]: Needs a conclusion

File: content/en/blog/2024/collecting-file-based-java-logs-with-opentelemetry.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
+## Problem
```
Comment (cyrille-leclerc) [inline content/en/blog/2024/collecting-file-based-java-logs-with-opentelemetry.md]: @zeitlinger shouldn't we remove the section titles?

File: content/en/blog/2024/collecting-file-based-java-logs-with-opentelemetry.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+
+The currently recommended way to send Java logs with OpenTelemetry is to use the OTLP protocol. However some use cases require outputting logs to files or stdout, due to organizational constraints or reliability practices that require usage of files for logs.
```
Comment (cyrille-leclerc) [inline content/en/blog/2024/collecting-file-based-java-logs-with-opentelemetry.md]:
```suggestion
The recommended way to send Java logs with OpenTelemetry is to use the OTLP protocol. However some use cases require outputting logs to files or stdout, due to organizational constraints or reliability practices that require usage of files for logs.
```

File: content/en/blog/2024/collecting-file-based-java-logs-with-opentelemetry.md [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+
+The deployment architecture looks like the following:
+
+![OTLP/JSON Architecture](collecting-file-based-java-logs-with-opentelemetry/otlpjson-architecture.png)
+
+## Set by step setup
+
+Steps
+
+* There is nothing to change in the code, continue to use your favorite logging library including getting the best of templated logs, mapped diagnostic context, and structured logging.
```
Comment (cyrille-leclerc) [inline content/en/blog/2024/collecting-file-based-java-logs-with-opentelemetry.md]: Should we have or not a trailing `.` on bullet points?

File: content/en/blog/2024/collecting-file-based-java-logs-with-opentelemetry.md [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+SeverityNumber: Info(9)
+Body: Str(Order order-12035 successfully placed)
+Attributes:
+     -> customerId: Str(cust-12345)
+     -> thread.id: Int(44)
+     -> thread.name: Str(grpc-default-executor-1)
+Trace ID: 42de1f0dd124e27619a9f3c10bccac1c
+Span ID: 270984d03e94bb8b
+Flags: 1
+	{"kind": "exporter", "data_type": "logs", "name": "debug"}
```
Comment (cyrille-leclerc) [inline content/en/blog/2024/collecting-file-based-java-logs-with-opentelemetry.md]:
```suggestion
     {"kind": "exporter", "data_type": "logs", "name": "debug"}
```

File: content/en/blog/2024/collecting-file-based-java-logs-with-opentelemetry.md [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+      IMPORTANT enable a console appender to troubleshoot cases were 
+      logs are missing in the OTel backend
+    -->
+  </root>
+</configuration>
+```
+
+## How to implement this when apps output logs to stdout on Kubernetes or Docker
+
+Supporting Kubernetes and Container specifics requires a standard parsing step in the pipeline without specific mapping configuration.
```
Comment (cyrille-leclerc) [inline content/en/blog/2024/collecting-file-based-java-logs-with-opentelemetry.md]: todo rephrase

File: content/en/blog/2024/collecting-file-based-java-logs-with-opentelemetry.md [hunk 7]
```diff
@@ -0,0 +1,228 @@
+## Problem
+
```
Comment (cyrille-leclerc) [inline content/en/blog/2024/collecting-file-based-java-logs-with-opentelemetry.md]: Needs an intro with the outline

**Comment by zeitlinger** [comment]
@cyrille-leclerc the pr doesn't target upstream

**Comment by zeitlinger** [comment]
I created https://github.com/cyrille-leclerc/opentelemetry.io/pull/3 - mostly with the help of copilot :smile:

File: content/en/blog/2024/collecting-file-based-java-logs-with-opentelemetry.md [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+     -> thread.name: Str(grpc-default-executor-1)
+Trace ID: 42de1f0dd124e27619a9f3c10bccac1c
+Span ID: 270984d03e94bb8b
+Flags: 1
+     {"kind": "exporter", "data_type": "logs", "name": "debug"}
+```
+
+* Verify the logs in the OpenTelemetry backend.
+* After the pipeline works end-to-end, ensure production readiness:
+  * Remove the "debug" exporter from the "logs" pipeline in the OTel Collector configuration.
```
Comment (cyrille-leclerc) [inline content/en/blog/2024/collecting-file-based-java-logs-with-opentelemetry.md]: `"` or `\`` to get colorization?

File: content/en/blog/2024/collecting-file-based-java-logs-with-opentelemetry.md [hunk 9]
```diff
@@ -0,0 +1,233 @@
...
+exporters:
+  debug:
+    verbosity: detailed
+  # Exporter to the OTLP backend like `otlphttp`
+  otlphttp:
+```
+
+## Conclusion
+
+This blog post showed how to collect file-based Java logs with OpenTelemetry. The solution is easy to set up and provides a turnkey solution for converting logs from frameworks like SLF4J/Logback or Log4j2 into OTel-compliant JSON logs on stdout with all resource and log attributes.
```
Comment (cyrille-leclerc) [inline content/en/blog/2024/collecting-file-based-java-logs-with-opentelemetry.md]: Should we mention that it's verbose and hat we think it's it good tradeoff?

Comment (zeitlinger) [reply content/en/blog/2024/collecting-file-based-java-logs-with-opentelemetry.md]: good idea

File: content/en/blog/2024/collecting-file-based-java-logs-with-opentelemetry.md (hunk 8 above)
Comment (zeitlinger) [reply content/en/blog/2024/collecting-file-based-java-logs-with-opentelemetry.md]: the latter

Comment (cyrille-leclerc) [inline content/en/blog/2024/collecting-file-based-java-logs-with-opentelemetry.md]:
```suggestion
  * Remove the `debug` exporter from the `logs` pipeline in the OTel Collector configuration.
```

File: content/en/blog/2024/collecting-file-based-java-logs-with-opentelemetry.md (hunk 9 above)
Comment (cyrille-leclerc) [reply content/en/blog/2024/collecting-file-based-java-logs-with-opentelemetry.md]:
```suggestion
This blog post showed how to collect file-based Java logs with OpenTelemetry. The solution is easy to set up and provides a turnkey solution for converting logs from frameworks like SLF4J/Logback or Log4j2 into OTel-compliant JSON logs on stdout with all resource and log attributes.
This JSON format is certainly verbose, but it generally has minimal impact on performances and offers a solid balance by providing highly contextualized logs that can be correlated with traces and metrics.
```

**Comment by zeitlinger** [comment]
@cyrille-leclerc why are you still working on this pr that targets your fork?

**Comment by cyrille-leclerc** [comment]
In fact i'm working on https://github.com/open-telemetry/opentelemetry.io/pull/5600/files but it's the same branch so this PR gets updated. I'll close this PR.

**Pull request closed without merging**